[docs] add 24.3.11.13 to releases page (#12912)
ydb-platform/ydb: ydb/docs/en/core/changelog-server.md

# {{ ydb-short-name }} Server changelog

## Version 24.3 {#24-3}

Release date: December 24, 2024.

### Functionality

> * Added a [cost-based optimizer](./concepts/optimizer#cost-based-query-optimizer) for complex queries on columnar tables. The cost-based optimizer considers a large number of alternative execution plans for each query and selects the best one based on the cost estimate for each option. For row-oriented tables, the optimizer can be enabled by configuring it in `table_service_config`. Currently, this optimizer only works with plans that contain [JOIN](./yql/reference/syntax/join) operations.

Reviewer: по умолчанию выключен?

Reviewer 2: По умолчанию на уровень 2 выставлен: оптимизатор включается на запросах где присутствует хоть одна колоночная таблица
https://ydb.tech/docs/ru/yql/reference/syntax/pragma#costbasedoptimizationlevel

Reviewer 3: Про строковые таблицы решили не писать, для запросов, в которых есть колоночные таблицы, оставили.

* Added support for [asynchronous replication](./concepts/async-replication), that allows synchronizing data between YDB databases in near real time. It can also be used for data migration between databases with minimal downtime for applications interacting with these databases.
* Extended [federated query](./concepts/federated_query/) capabilities to support new external data sources: MySQL, Microsoft SQL Server, and Greenplum.
* Published [documentation](./devops/manual/federated-queries/connector-deployment) on deploying YDB with [federated query](./concepts/federated_query/) functionality (manual setup).
* Added a new launch parameter `FQ_CONNECTOR_ENDPOINT` for YDB Docker containers that specifies an external data source connector address. Added support for TLS encryption for connections to the connector and the ability to expose the connector service port locally on the same host as the dynamic YDB node.
* Added support for auto-increment columns as part of a table's primary key.
* [Implemented](https://github.com/ydb-platform/ydb/pull/7150) [Change Data Capture (CDC)](./concepts/cdc) for synchronous secondary indexes.

> * Added delivery of [Count-min sketch](https://en.wikipedia.org/wiki/Count%E2%80%93min_sketch) statistics for [columnar tables](./concepts/datamodel/table#column-oriented-tables) to the query analyzer.

Reviewer: вообще она выключена и не работает AFAIK

Reviewer 4: Она выключена фиче флагом и разработка поставлена на паузу.

Reviewer 3: @azevaykin Саш, а есть инструкция, как включить фиче флаг для нее? Готовы заявить о ней в этой версии как об экспериментальной?

Reviewer 4: Недавно обнаружили, что тут явно недоделано. Но у команды сейчас фокус сменился. Поэтому лучше убрать из changelog.

https://github.com/ydb-platform/ydb/blob/4081b1fbb054156672e4c42b96015ff08cec2ac2/ydb/core/tx/columnshard/columnshard__statistics.cpp#L15

* Added audit logging for user login events in YDB, session termination events in the user interface, and backup/restore operations.

> * Initial version of the workload manager was implemented. It allows to create resource pools with CPU, memory and active queries count limits. Resource classifiers was implemented to assign queries to specific resource pool.

Reviewer 5: ```suggestion
* Initial version of the workload manager was implemented. It allows to create resource pools with CPU, memory and active queries count limits. Resource classifiers were implemented to assign queries to specific resource pool.
```

> * Initial version of the workload manager was implemented. It allows to create resource pools with CPU, memory and active queries count limits. Resource classifiers were implemented to assign queries to specific resource pool.

Reviewer: AFAIK тоже выключено

> * Added system view that allows you to get information about sessions installed from the database using a query.

Reviewer 5: ```suggestion
* Added a system view with information about sessions installed from the database using a query.
```

* Supported constant default values for columns.
* Enabled support for the RETURNING clause in queries.
* [Added](https://github.com/ydb-platform/ydb/pull/8708) start/end times and authors in the metadata for backup/restore operations from S3-compatible storage.
* Supported ACL for table backup/restore from S3-compatible storage.
* Included paths and decompression methods in query plans for reading from S3.
* Added new parsing options for timestamp/datetime fields when reading data from S3.

> * Enabled modification of retention periods for CDC topics.

Reviewer 5: ```suggestion
* Added support for changing retention periods in CDC topics.
```

> * Added support for changing record retention periods in [CDC]((./concepts/cdc)) topics.

Reviewer: я бы подвинул запись, чтобы сгруппировать с остальными строками про CDC

* Added support for the Decimal type in [partitioning keys](https://ydb.tech/docs/ru/dev/primary-key/column-oriented#klyuch-particionirovaniya).
* Improved diagnostics for storage issues in HealthCheck.

> * **_(Experimental)_** Added support for [VIEW](https://ydb.tech/docs/ru/concepts/datamodel/view).

Reviewer 5: ```suggestion
* **_(Experimental)_** Added support for [views](https://ydb.tech/docs/ru/concepts/datamodel/view).
```

> * **_(Experimental)_** Introduced an [auto-partitioning mode](./concepts/topic#autopartitioning) for topics, where partitions can dynamically split based on load while preserving message read-order and exactly-once guarantees. [Instructions](https://ydb.tech/docs/ru/yql/reference/syntax/alter-topic#autopartitioning) are provided for managing this mode.

Reviewer 5: ```suggestion
* **_(Experimental)_** Added an [auto-partitioning mode](./concepts/topic#autopartitioning) for topics, where partitions can dynamically split based on load while preserving message read-order and exactly-once guarantees. [Instructions](https://ydb.tech/docs/ru/yql/reference/syntax/alter-topic#autopartitioning) are provided for managing this mode.
```

> * **_(Experimental)_** Added an [auto-partitioning mode](./concepts/topic#autopartitioning) for topics, where partitions can dynamically split based on load while preserving message read-order and exactly-once guarantees.

Reviewer: А вот тут и ниже я удивлен. Автопаритионинг топиков гораздо менее экспериментальная штука, чем Workload manager и статистика.
Ниже про транзакции аналогично. В общем я бы поревьювил это "экспериментальное"

Reviewer 3: Поревьювили с Сергеем, Евгением и Алексеем, обновила.

* **_(Experimental)_** Added support for transactions involving [topics](https://ydb.tech/docs/ru/concepts/topic) and row-based tables, enabling transactional data transfer between tables and topics, or between topics, ensuring no data loss or duplication.
* **_(Experimental)_** Implemented [automatic index selection](https://ydb.tech/docs/ru/dev/secondary-indexes#avtomaticheskoe-ispolzovanie-indeksov-pri-vyborke) for queries, which can be enabled via the index_auto_choose_mode setting in table_service_config.
* **_(Experimental)_** Added support for write operations to columnar tables.

### YDB UI

> * Supported creation and [display](https://github.com/ydb-platform/ydb-embedded-ui/issues/782) of asynchronous replication instances.

Reviewer 5: ```suggestion
* Added support for creating and [viewing information on](https://github.com/ydb-platform/ydb-embedded-ui/issues/782) asynchronous replication instances.
```

> * [Added](https://github.com/ydb-platform/ydb-embedded-ui/issues/929) an indication for [auto-increment columns](./yql/reference/types/serial).

Reviewer 5: ```suggestion
* [Added](https://github.com/ydb-platform/ydb-embedded-ui/issues/929) an indicator for [auto-increment columns](./yql/reference/types/serial).
```

> * [Introduced](https://github.com/ydb-platform/ydb-embedded-ui/pull/1438) a page with information about [tablets](./concepts/glossary#tablet).

Reviewer 5: ```suggestion
* [Added](https://github.com/ydb-platform/ydb-embedded-ui/pull/1438) a page with information about [tablets](./concepts/glossary#tablet).
```

* Added a page with details about [distributed storage groups](./concepts/glossary#storage-group).

> * [Implemented](https://github.com/ydb-platform/ydb-embedded-ui/pull/1218) a setting to add tracing to all queries and display tracing results.

Reviewer 5: ```suggestion
* [Added](https://github.com/ydb-platform/ydb-embedded-ui/pull/1218) a setting to trace all queries and display tracing results.
```

* Enhanced the PDisk page with attributes, disk space consumption details, and a button to initiate [disk decommissioning](./devops/manual/decommissioning).
* Added information about currently running queries.

> * [Introduced](https://github.com/ydb-platform/ydb-embedded-ui/pull/1291) a row limit setting for query editor output and a notification when results exceed the limit.

Reviewer 5: ```suggestion
* [Added](https://github.com/ydb-platform/ydb-embedded-ui/pull/1291) a row limit setting for query editor output and a notification when results exceed the limit.
```

> * Added display of top CPU-consuming queries over the last hour.

Reviewer 5: ```suggestion
* Added an option to display top CPU-consuming queries over the last hour.
```

> * Enabled search on the history and saved queries pages.

Reviewer 5: ```suggestion
* Added a UI control to search the history and saved queries pages.
```

* Added the ability to cancel query execution.

> * [Introduced](https://github.com/ydb-platform/ydb-embedded-ui/issues/944) a shortcut to save queries from the editor.

Reviewer 5: ```suggestion
* [Added](https://github.com/ydb-platform/ydb-embedded-ui/issues/944) a shortcut to save queries in the editor.
```

* Added a display of the number of rows retrieved for all queries.
* [Separated](https://github.com/ydb-platform/ydb-embedded-ui/pull/1422) donor disks from other disks in the UI.

> * Supported InterruptInheritance ACL and improved visualization of active ACLs.

Reviewer 5: ```suggestion
* Added support for InterruptInheritance ACL and improved visualization of active ACLs.
```

* [Added](https://github.com/ydb-platform/ydb-embedded-ui/pull/889) a display of the current UI version.

> * Introduced functionality to enable/disable experimental database features.

Reviewer 5: ```suggestion
* Added support for enabling/disabling experimental database features.
```

* Enhanced the Hive tablet balancing algorithm.
* Accelerated recovery of tables with secondary indexes from backups.
* Optimized Interconnect throughput.

### Bug fixes

* [Fixed](https://github.com/ydb-platform/ydb/pull/6850) an issue that caused databases with a large number of tables or partitions to become non-functional during restoration from a backup. Now, if database size limits are exceeded, the restoration operation will fail, but the database will remain operational.
* Implemented a mechanism to forcibly trigger background [compaction](./concepts/glossary#compaction) when discrepancies between the data schema and stored data are detected in [DataShard](./concepts/glossary#data-shard). This resolves a rare issue with delays in schema changes.
* [Resolved](https://github.com/ydb-platform/ydb/pull/10447) duplication of authentication tickets, which led to an increased number of requests to authentication providers.
* [Fixed](https://github.com/ydb-platform/ydb/pull/9377) an invariant violation issue during the initial scan of CDC.
* [Prohibited](https://github.com/ydb-platform/ydb/pull/9446) schema changes for backup tables.

> * [Fixed](https://github.com/ydb-platform/ydb/pull/9509) a hang during the initial scan of CDC when the table is frequently updated.

Reviewer 5: ```suggestion
* [Fixed](https://github.com/ydb-platform/ydb/pull/9509) an issue with an initial scan freezing during CDC when the table is frequently updated.
```
